ocaml/xapi/server_helpers.ml

```diff
@@ -75,19 +75,21 @@ let exec ?marshaller ?f_forward ~__context f =
           (* use the forwarding layer (NB this might make a local call ultimately) *)
           forward ~local_fn:f ~__context
     in
-    begin match marshaller with
+    (* For forwarded task, we should not complete it here, the server which forward the task will complete it *)
+    if not (Context.forwarded_task __context) then begin
+      match marshaller with
       | None    -> TaskHelper.complete ~__context None
       | Some fn -> TaskHelper.complete ~__context (Some (fn result))
```

Comment:
@robhoes I have some concern about this. After this change, the result in the slave won't be updated to the task result, but instead will update with the execution result in the master, will it bring some other issues?

Reply:
I should hope that the result of the forwarded call is already always returned (unaltered) by the master to the original caller, especially for synchronous calls. So I think this is fine.